src/Package.js

```diff
@@ -96,6 +96,9 @@ export default class Package {
     // check if semantic versions are compatible
     if (semver.satisfies(actualVersion, expectedVersion)) {
       return true;
+    // or the dependency is specified as a relative folder
+    } else if (/^file:\.\.\//.test(dependency.version)) {
```

Comment:
Seems not too fancy?  The [supported dependency forms](https://docs.npmjs.com/files/package.json#local-paths) are pretty limited.

I guess we can start with this, and I'll file a ticket for improving support.